dashboard/app/models/pd/application/application_base.rb

```diff
@@ -222,36 +174,12 @@ def self.statuses
     # This is equivalent to
     #   validates_inclusion_of :status, in: statuses
     # but it will work with derived classes that override statuses
-    validate :status_is_valid_for_application_type
     def status_is_valid_for_application_type
       unless status.nil? || self.class.statuses.include?(status)
         errors.add(:status, 'is not included in the list.')
       end
     end
 
-    enum course: %w(
-      csf
-      csd
-      csp
-      csa
-    ).index_by(&:to_sym).freeze
-
-    COURSE_NAME_MAP = {
-      csp: Pd::Workshop::COURSE_CSP,
-      csd: Pd::Workshop::COURSE_CSD,
-      csf: Pd::Workshop::COURSE_CSF,
-      csa: Pd::Workshop::COURSE_CSA
-    }
-
-    belongs_to :user
-    belongs_to :regional_partner
-
-    validates_presence_of :user_id, unless: proc {|application| application.application_type == PRINCIPAL_APPROVAL_APPLICATION}
-    validates_inclusion_of :application_type, in: APPLICATION_TYPES
-    validates_inclusion_of :application_year, in: APPLICATION_YEARS
-    validates_presence_of :type
-    validates_presence_of :status, unless: proc {|application| application.application_type == PRINCIPAL_APPROVAL_APPLICATION}
```

Comment:
Moved these up to the top of the file as well. Tbh, I totally missed the status validation for a while, not expecting it to be down her ...